### PR TITLE
lgtm plugin should use trigger.TrustedUser

### LINKED ISSUE #13740: lgtm plugin should use trigger.TrustedUser
The lgtm plugin currently does a `IsCollaborator()` check to determine if a user can lgtm a PR or not. Which means that even if `only_org_members: true` is defined in the `trigger` config, outside collaborators are able to lgtm PRs.

To fix this, we should use `trigger.TrustedUser()` to determine whether a user is trusted or not.

Additionally, if `skip_collaborators: true` is configured, we can then skip the `trigger.TrustedUser()` check.

/sig contributor-experience
/cc @cblecker @spiffxp @alvaroaleman @stevekuznetsov 
related: https://github.com/kubernetes/test-infra/issues/13002

Happy to work on this if this makes sense.

### PR DESCRIPTION
**Pull request by medmedchiheb**
change lgtm plugin to use trigger.TrustedUser

/cc @nikhita @stevekuznetsov 

Fixes #13740

### CODE REVIEW

**Comment by nikhita** [comment]
@cblecker I believe you had some concerns around this?

File: prow/plugins/lgtm/lgtm.go [hunk 1]
```diff
@@ -288,17 +289,19 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 	}
 
 	// check if skip collaborators is enabled for this org/repo
-	skipCollaborators := skipCollaborators(config, org, repoName)
+	skipCollaborators := skipCollaborators(config, org, repoName)
```
Comment (cblecker) [inline prow/plugins/lgtm/lgtm.go]: nit: extra white space?

File: prow/plugins/lgtm/lgtm.go [hunk 2]
```diff
@@ -288,17 +289,19 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
...
+	t := gc.PluginConfig.TriggerFor(org, repoName)
+	isTrustedUser , err :=  trigger.TrustedUser(gc, t.OnlyOrgMembers, t.TrustedOrg, author, org, repoName)
 
-	// check if the commentor is a collaborator
-	isCollaborator, err := gc.IsCollaborator(org, repoName, author)
 	if err != nil {
 		log.WithError(err).Error("Failed to check if author is a collaborator.")
 		return err // abort if we can't determine if commentor is a collaborator
-	}
+	}
```
Comment (cblecker) [inline prow/plugins/lgtm/lgtm.go]: nit: extra white space?

File: prow/plugins/lgtm/lgtm.go [hunk 3]
```diff
@@ -288,17 +289,19 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 	}
 
 	// check if skip collaborators is enabled for this org/repo
-	skipCollaborators := skipCollaborators(config, org, repoName)
+	skipCollaborators := skipCollaborators(config, org, repoName)	
+
+	// check if the commentor is a trusted user
+	t := gc.PluginConfig.TriggerFor(org, repoName)
+	isTrustedUser , err :=  trigger.TrustedUser(gc, t.OnlyOrgMembers, t.TrustedOrg, author, org, repoName)
```
Comment (cblecker) [inline prow/plugins/lgtm/lgtm.go]:
```suggestion
	isTrustedUser, err :=  trigger.TrustedUser(gc, t.OnlyOrgMembers, t.TrustedOrg, author, org, repoName)
```

**Comment by cblecker** [comment]
I want to rip out the trusted org bits, I think. I know this is confusing if the configuration varies, but the is collab and trusteduser code paths result in different values in some cases.

**Pull request closed without merging**